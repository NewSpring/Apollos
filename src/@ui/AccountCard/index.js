import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import { Platform } from 'react-native';
import { startCase, toLower } from 'lodash';

import { withIsLoading } from '@ui/isLoading';
import styled from '@ui/styled';
import Card, { CardContent } from '@ui/Card';
import SideBySideView from '@ui/SideBySideView';
import FlexedView from '@ui/FlexedView';
import { H5 } from '@ui/typography';
import CategoryLabel from '@ui/CategoryLabel';
import Icon from '@ui/Icon';
import last4 from '@utils/last4';
import { withTheme } from '@ui/theme';

const enhance = compose(
  setPropTypes({
    title: PropTypes.string.isRequired,
    isLoading: PropTypes.bool,
    accountNumber: PropTypes.number,
    accountType: PropTypes.number,
    iconSize: PropTypes.number,
  }),
  defaultProps({
    isLoading: false,
    accountNumber: '',
    accountType: 'creditCard',
    iconSize: undefined,
  }),
  withIsLoading,
  withTheme(({ theme, ...otherProps }) => ({
    iconSize: otherProps.iconSize || theme.helpers.rem(1),
  })),
  pure,
);

const HorizontalLayout = styled({
  alignItems: 'center',
  minHeight: 110, // kind of the best middle ground for various title lengths.
})(SideBySideView);

const LeftColumn = styled({
  flex: 1.66,
})(CardContent);

const RightColumn = styled(({ theme }) => ({
  paddingRight: theme.sizing.baseUnit,
  alignItems: 'flex-end',
  justifyContent: 'center',
  ...Platform.select({
    web: {
      position: 'relative',
    },
  }),
}))(FlexedView);

const AccountCard = enhance(({
  title,
  isLoading,
  accountNumber,
  accountType,
  iconSize,
  ...otherProps
}) => (
  <Card isLoading={isLoading} {...otherProps}>
    <HorizontalLayout>
      <LeftColumn>
        <H5>{startCase(toLower(title))}</H5>
        <CategoryLabel
          label={last4(accountNumber)}
          icon={accountType === 'bankAccount' ? 'bank' : 'credit'}
          isLoading={isLoading}
        />
      </LeftColumn>

      <RightColumn>
        <Icon name="ArrowNext" size={iconSize} />
      </RightColumn>
    </HorizontalLayout>
  </Card>
));

export default AccountCard;
