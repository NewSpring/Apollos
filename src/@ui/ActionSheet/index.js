export { ActionSheetProvider, connectActionSheet } from '@expo/react-native-action-sheet';
