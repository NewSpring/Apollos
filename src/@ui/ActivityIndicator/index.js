import { ActivityIndicator as NativeActivityIndicator, StyleSheet } from 'react-native';
import styled from '@ui/styled';

const ActivityIndicator = styled({
  flex: 1,
  ...StyleSheet.absoluteFillObject,
  alignSelf: 'center',
}, 'ActivityIndicator')(NativeActivityIndicator);

export {
  ActivityIndicator as default,
  NativeActivityIndicator as InlineActivityIndicator,
};
