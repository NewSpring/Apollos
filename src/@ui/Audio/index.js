import React, { Component } from 'react';
import { Audio as ExpoAudio } from 'expo';
import PropTypes from 'prop-types';
import { Animated, View } from 'react-native';
import AudioPlay from './AudioPlay';
import AudioPause from './AudioPause';
import AudioSeeker from './AudioSeeker';

export default class Audio extends Component {
  static Play = AudioPlay;
  static Pause = AudioPause;
  static Seeker = AudioSeeker;

  static propTypes = {
    source: PropTypes.string,
    onReady: PropTypes.func,
    onError: PropTypes.func,
    onPlaybackReachedEnd: PropTypes.func,
    onPlay: PropTypes.func,
    onPause: PropTypes.func,
    onStop: PropTypes.func,
    onSeek: PropTypes.func,
    onSeeking: PropTypes.func,
    children: PropTypes.node,
    isPlaying: PropTypes.bool,
    style: PropTypes.any, // eslint-disable-line
  };

  static defaultProps = {
    onReady() {},
    onError(err) {
      // eslint-disable-next-line no-console
      console.error(err.message);
    },
    onPlaybackReachedEnd() {},
    onPlay() {},
    onPause() {},
    onStop() {},
    onSeek() {},
    onSeeking() {},
    children: null,
    isPlaying: false,
  };

  static childContextTypes = {
    play: PropTypes.func,
    stop: PropTypes.func,
    pause: PropTypes.func,
    seek: PropTypes.func,
    progress: PropTypes.object,
    positionMillis: PropTypes.object,
    seekingHandler: PropTypes.func,
  };

  getChildContext = () => ({
    play: this.play,
    stop: this.stop,
    pause: this.pause,
    seek: this.seek,
    progress: this.progressDriver,
    positionMillis: this.positionMillisDriver,
    seekingHandler: this.handleSeeking,
  });

  componentWillMount() {
    ExpoAudio.setAudioModeAsync({
      allowsRecordingIOS: false,
      interruptionModeIOS: Audio.INTERRUPTION_MODE_IOS_DO_NOT_MIX,
      playsInSilentModeIOS: true,
      shouldDuckAndroid: true,
      interruptionModeAndroid: Audio.INTERRUPTION_MODE_ANDROID_DO_NOT_MIX,
    });

    this.loadSource();
  }

  componentDidUpdate({ source, isPlaying }) {
    if (this.props.source !== source) {
      this.loadSource();
    } else if (this.props.isPlaying !== isPlaying) {
      if (this.props.isPlaying) this.play();
      if (!this.props.isPlaying) this.pause();
    }
  }

  componentWillUnmount() {
    if (this.sound) this.sound.unloadAsync();
  }

  onPlaybackStatusUpdate = (soundStatus) => {
    this.positionMillisDriver.setValue(soundStatus.positionMillis);
    this.progressDriver.setValue((soundStatus.positionMillis / this.duration) || 0);

    if (soundStatus.didJustFinish) {
      this.props.onPlaybackReachedEnd(this.sound);
    }
  }

  positionMillisDriver = new Animated.Value(0);
  progressDriver = new Animated.Value(0);
  duration = 0;
  positionListener = undefined;
  previousSoundStatus = undefined;
  isReady = false;
  playState = this.props.isPlaying;

  play = async () => {
    if (this.sound) {
      try {
        await this.sound.playAsync();
        this.props.onPlay();
      } catch (err) {
        this.props.onError(err);
      }
    }
  }

  pause = async () => {
    if (this.sound) {
      try {
        await this.sound.pauseAsync();
        this.props.onPause();
      } catch (err) {
        this.props.onError(err);
      }
    }
  }

  stop = async () => {
    try {
      this.sound.stopAsync();
      this.props.onStop();
    } catch (err) {
      this.props.onError(err);
    }
  }

  seek = (percentageOfSong) => {
    const positionInMillis = this.duration * percentageOfSong;
    this.sound.setPositionAsync(positionInMillis);
    this.props.onSeek(positionInMillis);
  }

  handleSeeking = (percentageOfSong) => {
    const positionInMillis = this.duration * percentageOfSong;
    this.props.onSeeking(positionInMillis);
  }

  loadSource = async () => {
    const {
      source,
      onReady,
      onError,
    } = this.props;

    if (!source) return;
    const uri = source.replace(/^http:\/\/|^\/\//i, 'https://');

    try {
      if (this.sound && this.soundLoaded === this.sound) {
        await this.sound.unloadAsync();
      }

      const sound = this.sound = await new ExpoAudio.Sound(); // eslint-disable-line
      const soundStatus = await this.sound.loadAsync({ uri });
      this.soundLoaded = this.sound;

      if (sound === this.sound) { // since the above calls are promise base, there's a chance that
      // the current sound changes before this sound is loaded (pressing skip button quickly).
      // We don't want to start playing this sound in that scenario :)
        this.duration = soundStatus.durationMillis;
        this.sound.setOnPlaybackStatusUpdate(this.onPlaybackStatusUpdate);
        if (this.props.isPlaying) await this.play();
      }

      onReady();
    } catch (err) {
      onError(err);
    }
  }

  render() {
    return <View style={this.props.style}>{this.props.children}</View>;
  }
}
