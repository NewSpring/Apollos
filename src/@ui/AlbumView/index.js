import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';
import { View, Platform, StyleSheet } from 'react-native';
import PaddedView from '@ui/PaddedView';
import ProgressiveImage from '@ui/ProgressiveImage';
import { H5, H7 } from '@ui/typography';
import { withThemeMixin } from '@ui/theme';
import styled from '@ui/styled';

const Container = styled(({ theme }) => ({
  backgroundColor: theme.colors.background.default,
}), 'AlbumView.Container')(View);

const InfoWrapper = styled({
  flexDirection: 'row', alignItems: 'stretch',
}, 'AlbumView.InfoWrapper')(PaddedView);

const AlbumArt = styled({
  position: 'relative',
  width: '33%',
  aspectRatio: 1,
  ...Platform.select({
    web: {
      height: 0,
      paddingTop: '100%',
    },
  }),
}, 'AlbumView.AlbumArt')(ProgressiveImage);

const BlurredImage = styled(({ theme }) => ({
  ...StyleSheet.absoluteFillObject,
  opacity: theme.alpha.low,
}), 'AlbumView.BlurredImage')(ProgressiveImage);

const TitleWrapper = styled({
  width: '66%',
  justifyContent: 'space-around',
}, 'AlbumView.TitleWrapper')(PaddedView);

const Title = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Title')(H5);

const Artist = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Artist')(H7);

const enhance = compose(
  withThemeMixin({ type: 'dark' }),
  pure,
  setPropTypes({
    title: PropTypes.string,
    artist: PropTypes.string,
    blurredImage: ProgressiveImage.propTypes.source,
    albumImage: ProgressiveImage.propTypes.source,
    isLoading: PropTypes.bool,
  }),
);

const AlbumView = enhance(({
  title, artist, blurredImage, albumImage, isLoading = false,
}) => (
  <Container>
    {(blurredImage) ? <BlurredImage source={blurredImage} /> : null}
    <InfoWrapper>
      <AlbumArt isLoading={isLoading} source={albumImage} />
      <TitleWrapper>
        <Title>{title}</Title>
        <Artist>{artist}</Artist>
      </TitleWrapper>
    </InfoWrapper>
  </Container>
));

export default AlbumView;
