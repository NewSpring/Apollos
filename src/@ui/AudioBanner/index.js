import React from 'react';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import styled from '@ui/styled';
import PaddedView from '@ui/PaddedView';
import Icon from '@ui/Icon';
import { H7 } from '@ui/typography';

const enhance = compose(
  pure,
  setPropTypes({
    titleText: PropTypes.string,
  }),
  defaultProps({
    titleText: 'Listen to audio',
  }),
);

const Banner = styled({
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',
})(PaddedView);

const AudioBanner = enhance(({ titleText }) => (
  <Banner>
    <View>
      <H7>{titleText}</H7>
    </View>
    <Icon name={'audio'} />
  </Banner>
));

export default AudioBanner;
