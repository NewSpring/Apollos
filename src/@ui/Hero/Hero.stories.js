import React from 'react';
import { StyleSheet } from 'react-native';
import { storiesOf } from '@storybook/react-native';
import { H1 } from '@ui/typography';
import FlexedView from '@ui/FlexedView';
import Video from '@ui/VideoPlayer';

import Hero, { BackgroundImage } from './';

const BackgroundVideo = () => (
  <Video
    src="https://s3.amazonaws.com/ns.images/newspring/fpo/HomepageVideo_ForExport_V3-Web_Hero_2_000kbps.mp4"
    posterSource="https://dg0ddngxdz549.cloudfront.net/images/cached/images/remote/http_s3.amazonaws.com/ns.images/newspring/homepage/hero_poster_2x1_1700_850_90_c1.jpg"
    useNativeControls={false}
    shouldPlay
    isLooping
  />
);

storiesOf('Hero', module)
  .addDecorator(renderStory => (
    <FlexedView style={StyleSheet.absoluteFill}>{renderStory()}</FlexedView>
  ))
  .add('with image background', () => (
    <Hero
      brandText="My Cool Church"
      backgroundOpacity={0.5}
      background={<BackgroundImage source="https://picsum.photos/600/400/?image=63" />}
    >
      <H1>Hello!</H1>
    </Hero>
  ))
  .add('with video background', () => (
    <Hero
      brandText="My Cool Church"
      backgroundOpacity={0.5}
      background={<BackgroundVideo />}
    >
      <H1>Hello!</H1>
    </Hero>
  ))
  .add('with solid background', () => (
    <Hero
      brandText="My Cool Church"
      backgroundColor="salmon"
    >
      <H1>Hello!</H1>
    </Hero>
  ));

