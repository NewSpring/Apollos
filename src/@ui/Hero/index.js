import React from 'react';
import { Platform, View, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import styled from '@ui/styled';
import ConnectedImage from '@ui/ConnectedImage';
import PaddedView from '@ui/PaddedView';
import FlexedView from '@ui/FlexedView';
import { withThemeMixin } from '@ui/theme';
import { H4 } from '@ui/typography';

const BackgroundContainer = styled(({ theme, backgroundColor }) => ({
  backgroundColor: backgroundColor || theme.colors.darkPrimary,
  ...StyleSheet.absoluteFillObject,
}))(View);

const BackgroundFade = styled(({ opacity }) => ({
  opacity: opacity || 0.8,
  alignItems: 'stretch',
  ...StyleSheet.absoluteFillObject,
}))(View);

const Content = styled({
  alignItems: 'center',
  justifyContent: 'center',
})(PaddedView);

const Brand = styled(({ theme }) => ({
  color: theme.colors.white,
  position: 'absolute',
  bottom: theme.sizing.baseUnit,
  right: theme.sizing.baseUnit,
  ...Platform.select({
    web: {
      right: 0,
      bottom: 0,
      'text-shadow': '0 0 10px #858585', // todo
      transform: 'rotate(-90deg) translateY(300%)',
      'transform-origin': '0% 0%',
    },
  }),
}))(H4);

const Container = styled({
  overflow: 'hidden',
  alignItems: 'center',
  justifyContent: 'center',
})(FlexedView);

export const BackgroundImage = styled({
  width: '100%',
  height: '100%',
  resizeMode: 'cover',
})(ConnectedImage);

const Hero = withThemeMixin({ type: 'dark' })(({
  brandText = 'NEWSPRING',
  backgroundColor,
  backgroundOpacity,
  background,
  children,
  style,
  contentContainerStyle = {},
}) => (
  <Container style={style}>
    <BackgroundContainer backgroundColor={backgroundColor}>
      <BackgroundFade opacity={backgroundOpacity}>{background}</BackgroundFade>
    </BackgroundContainer>
    <Content style={contentContainerStyle}>
      {children}
    </Content>
    <Brand>{brandText}</Brand>
  </Container>
));

Hero.propTypes = {
  brandText: PropTypes.string,
  backgroundColor: PropTypes.string,
  backgroundOpacity: PropTypes.number,
  background: PropTypes.node,
  children: PropTypes.node,
  style: PropTypes.any, // eslint-disable-line
  contentContainerStyle: PropTypes.any, // eslint-disable-line
};

export default Hero;
