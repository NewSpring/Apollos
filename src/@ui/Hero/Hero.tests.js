import React from 'react';
import renderer from 'react-test-renderer';
import Providers from '@ui/TestProviders';
import { H1 } from '@ui/typography';

import Hero, { BackgroundImage } from './';

describe('The Hero component', () => {
  it('should render with background image', () => {
    const tree = renderer.create(
      <Providers>
        <Hero
          brandText="My Cool Church"
          backgroundOpacity={0.5}
          background={<BackgroundImage source="https://picsum.photos/600/400/?image=63" />}
        >
          <H1>Hello!</H1>
        </Hero>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with arbitrary background', () => {
    const tree = renderer.create(
      <Providers>
        <Hero
          background={<H1>This would be dumb, but should work fine!</H1>}
        >
          <H1>Hello!</H1>
        </Hero>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with solid background', () => {
    const tree = renderer.create(
      <Providers>
        <Hero
          brandText="My Cool Church"
          backgroundColor="salmon"
        >
          <H1>Hello!</H1>
        </Hero>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
