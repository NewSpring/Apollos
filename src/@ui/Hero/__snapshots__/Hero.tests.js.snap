// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Hero component should render with arbitrary background 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "overflow": "hidden",
    }
  }
>
  <View
    backgroundColor={undefined}
    style={
      Object {
        "backgroundColor": "#303030",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      opacity={undefined}
      style={
        Object {
          "alignItems": "stretch",
          "bottom": 0,
          "left": 0,
          "opacity": 0.8,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Colfax-Bold",
            "fontSize": 52.2,
            "lineHeight": 54.81,
            "paddingTop": 11.7,
          }
        }
      >
        This would be dumb, but should work fine!
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Colfax-Bold",
          "fontSize": 52.2,
          "lineHeight": 54.81,
          "paddingTop": 11.7,
        }
      }
    >
      Hello!
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "bottom": 20,
        "color": "#ffffff",
        "fontFamily": "Colfax-Bold",
        "fontSize": 25.2,
        "lineHeight": 31.28,
        "paddingBottom": 3.71,
        "paddingTop": 5.04,
        "position": "absolute",
        "right": 20,
      }
    }
  >
    NEWSPRING
  </Text>
</View>
`;

exports[`The Hero component should render with background image 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "overflow": "hidden",
    }
  }
>
  <View
    backgroundColor={undefined}
    style={
      Object {
        "backgroundColor": "#303030",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      opacity={0.5}
      style={
        Object {
          "alignItems": "stretch",
          "bottom": 0,
          "left": 0,
          "opacity": 0.5,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#858585",
            "width": "100%",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Colfax-Bold",
          "fontSize": 52.2,
          "lineHeight": 54.81,
          "paddingTop": 11.7,
        }
      }
    >
      Hello!
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "bottom": 20,
        "color": "#ffffff",
        "fontFamily": "Colfax-Bold",
        "fontSize": 25.2,
        "lineHeight": 31.28,
        "paddingBottom": 3.71,
        "paddingTop": 5.04,
        "position": "absolute",
        "right": 20,
      }
    }
  >
    My Cool Church
  </Text>
</View>
`;

exports[`The Hero component should render with solid background 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "overflow": "hidden",
    }
  }
>
  <View
    backgroundColor="salmon"
    style={
      Object {
        "backgroundColor": "salmon",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      opacity={undefined}
      style={
        Object {
          "alignItems": "stretch",
          "bottom": 0,
          "left": 0,
          "opacity": 0.8,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Colfax-Bold",
          "fontSize": 52.2,
          "lineHeight": 54.81,
          "paddingTop": 11.7,
        }
      }
    >
      Hello!
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "bottom": 20,
        "color": "#ffffff",
        "fontFamily": "Colfax-Bold",
        "fontSize": 25.2,
        "lineHeight": 31.28,
        "paddingBottom": 3.71,
        "paddingTop": 5.04,
        "position": "absolute",
        "right": 20,
      }
    }
  >
    My Cool Church
  </Text>
</View>
`;
